### PR TITLE
fixes for setting layout and building problem fields

### PR DESCRIPTION
**Pull request by cderb**
Added tensor constructor with layout field.
Correct BuildConvProblem for bwd and wrw configs.

### CODE REVIEW

**Comment by JehandadKhan** [comment]
@bghimireamd Please take some time to review this PR.

**Comment by bghimireamd** [comment]
CI is failing on some minor issue. Once CI passed we can merge.

File: src/include/tensor.hpp [hunk 1]
```diff
@@ -198,6 +198,22 @@ struct tensor
         clGetCommandQueueInfo(q, CL_QUEUE_CONTEXT, sizeof(cl_context), &ctx, nullptr);
 #endif
     }
+    template <typename U>
+    tensor(accelerator_stream _q,
+           std::string layout,
```
Comment (bghimireamd) [inline src/include/tensor.hpp]:
```suggestion
           const std::string& layout,
```

File: src/include/tensor.hpp [hunk 2]
```diff
@@ -198,6 +198,22 @@ struct tensor
...
+           std::vector<U>& _plens,
```
Comment (bghimireamd) [inline src/include/tensor.hpp]:
```suggestion
           const std::vector<U>& _plens,
```

**Review by bghimireamd — APPROVED** [review]

**Pull request merged**